README.md

```diff
@@ -164,6 +164,41 @@ Please set the environmental variable `TF_LOG` to one of the log levels `TRACE`,
 
 Set the environmental variable `TMC_MODE` to `DEV` to capture more granular logs.
 
+## Connect the VSCode debugger
+1. Create `./.vscode/launch.json`
+    ```json5
+    {
+          "version": "0.2.0",
+          "configurations": [
+              {
+                  "name": "Debug Terraform Provider",
+                  "type": "go",
+                  "request": "launch",
+                  "mode": "debug",
+                  // this assumes your workspace is the root of the repo
+                  "program": "${workspaceFolder}",
+                  "env": {},
+                  "args": [
+                      "-debug",
+                  ]
+              }
+          ]
+      }
+    ```
+2. Click on **"Run and Debug"** option in VSCode, This will open a panel on the left side of the editor. Here, you can see a list of configurations for debugging different languages and tools. Find the one that says **"Debug Terraform Provider"** and click on it. This will launch the debugger and attach it to your provider process. You can now set breakpoints, inspect variables, and step through your code as usual.
+3. Check the **"DEBUG CONSOLE"** tab, there you will find the value of `TF_REATTACH_PROVIDERS`, which is a special environment variable that tells Terraform how to connect to the provider's plugin process. You need to set this variable in your shell before running any Terraform commands. For example, you can use the export command as shown below:
+    <br/>
+    ```sh
+    # Set TF_REATTACH_PROVIDERS as environment variable.
+    export TF_REATTACH_PROVIDERS='{"registry.terraform.io/my-org/my-provider":{"Protocol":"grpc","Pid":3382870,"Test":true,"Addr":{"Network":"unix","String":"/tmp/plugin713096927"}}}'
```

Comment:
Can we point it to our provider?

Reply:
Updated, but the `Pid` and the `Addr.String` will be different for each debug session.